docs/antrea-proxy.md

```diff
@@ -42,14 +42,20 @@ the introduction of `proxyAll`, Antrea relied on userspace kube-proxy, which is
 no longer actively maintained by the K8s community and is slower than other
 kube-proxy backends.
 
-Note that on Linux, even when `proxyAll` is enabled, kube-proxy will usually
-take priority and will keep handling NodePort Service traffic (unless the source
-is a Pod, which is pretty unusual as Pods typically access Services by
-ClusterIP). This is because kube-proxy rules typically come before the rules
-installed by AntreaProxy to redirect traffic to OVS. When kube-proxy is not
-deployed or is removed from the cluster, AntreaProxy will then handle all
+Note that on Linux, before Antrea v2.1, when `proxyAll` is enabled, kube-proxy
+will usually take priority and will keep handling NodePort Service traffic
+(unless the source is a Pod, which is pretty unusual as Pods typically access
+Services by ClusterIP). This is because kube-proxy rules typically come before
+the rules installed by AntreaProxy to redirect traffic to OVS. When kube-proxy
+is not deployed or is removed from the cluster, AntreaProxy will then handle all
 Service traffic.
 
+It's worth noting that starting with Antrea v2.1, when only `proxyAll` is enabled,
+even if kube-proxy is present, AntreaProxy is capable of handing all types of
+Service traffic except for that of the kube Service,. This is accomplished by
+prioritizing the rules installed by AntreaProxy redirecting Service traffic to OVS
+over those installed by kube-proxy.
+
```

Comment:
From 2.1 , If proxyAll is enabled, is there any way if user would still have control to continue to allow kube-proxy to handle all types of service traffic instead of letting Antrea-proxy to redirect all service types traffic to OVS by default?

Reply:
Is it a meaningful case? With proxyAll enabled, but AntreaProxy doesn't take effect.

Reply:
With proxyAll enabled, Antrea-proxy will handle all service traffic by default through OVS, effectively preventing kube-proxy from managing any service traffic due to the deprioritization of kube-proxy rules, even if kube-proxy is present.
This setup is essentially equivalent to having only Antrea-Proxy and removing kube-proxy, as Antrea-Proxy takes over all service traffic management.
However, we might want to consider allowing users to still have the option for kube-proxy to handle service traffic. Adding an additional flag could provide users the flexibility to route all service traffic through kube-proxy, even when proxyAll is enabled. This would help maintain backward compatibility and offer more control over service traffic handling !

Reply:
I would just disable `proxyAll` to achieve that.

Reply:
Usually proxyAll was introduced for the possibility to remove kube-proxy to avoid running kube-proxy pods in the cluster and use less resources ! But with this change we like to have kube-proxy pods still run in cluster but allow antrea-proxy to take over.

Reply:
You could take a look at #6232, the motivation of the PR.  Have you got any ideas about the case you described? Introducing another option, like `proxyAllWithKubeProxy`, to be compatible with the case?

Reply:
I don't really get what was actually suggested here, but we should NOT definitely add an option to let kube-proxy handle the traffic, that's what `antreaProxy.enable` and `antreaProxy.proxyAll` do exactly.

Reply:
When proxyAll is enabled, it does suggest that both AntreaProxy and kube-proxy are present in the system. However, the intention behind proxyAll is to allow AntreaProxy to handle all Service traffic, potentially removing the need for kube-proxy.

can we use antreaProxy.enable with proxyAll set to true and add an option to disable kube-proxy in antrea ? , where users want to simplify their proxy setup and rely solely on AntreaProxy. This approach would streamline the configuration and reduce the overhead of removing kube-proxy manually !

Here's how this could be implemented:

Configuration Parameter: Introduce a new configuration parameter, such as antreaProxy.disableKubeProxy

Automated Removal: If antreaProxy.disableKubeProxy is set to true, it could remove kube-proxy from the cluster.

Reconciliation: Instead of relying on overrides, the system could perform reconciliation(if required) to ensure that the desired state (AntreaProxy enabled, kube-proxy disabled) is maintained across the cluster.

If we are on same page we can do by opening an another issue and in another PR !

Reply:
This is out of scope of Antrea again. None users would expect a network plugin to manage other Pods' lifecycle in their clusters, and we would never request that power for Antrea. I would suggest you to read the issue which the PR is created for first to understand what problem the PR tries to resolve.

I would not suggest to open an issue for making antrea be able to remove kube-proxy, for the above reasons, it's not very worth broader discussion IMO.

Reply:
Removing kube-proxy is currently supported https://github.com/antrea-io/antrea/blob/main/docs/antrea-proxy.md#removing-kube-proxy

Reply:
@rajnkamr could you please review hongliang and quan's comments carefully and double check the docs before you left comments and start a discussion?  Sometime it will save teams from extra efforts. Thanks.
Btw, `Removing kube-proxy` section in the document is totally a different thing with the one you bring up in the thread.

Reply:
@luolanzone , the discussion was more related on doc part for user simplification of usage, Sure, we can take it separately .